react-native/react/reducers/tracker.js

```diff
@@ -29,67 +28,67 @@ type State = {
   lastTrack: ?TrackSummary
 }
 
+type State = {
+  serverStarted: boolean,
+  trackers: {[key: string]: TrackerState}
+}
+
 const initialProofState = checking
 
 const initialState: State = {
   serverStarted: false,
-  serverActive: false,
-  username: null,
-  proofState: initialProofState,
-  shouldFollow: true,
-  proofs: [],
-  reason: '', // TODO: get the reason
-  closed: true,
-  lastTrack: null,
-  userInfo: {
-    fullname: 'TODO: get this information',
-    followersCount: -1,
-    followingCount: -1,
-    followsYou: false,
-    avatar: null,
-    location: '' // TODO: get this information
+  trackers: {}
+}
+
+function initialTrackerState (): TrackerState {
+  return {
+    serverActive: false,
+    username: null,
+    proofState: initialProofState,
+    shouldFollow: true,
+    proofs: [],
+    reason: '', // TODO: get the reason
+    closed: true,
+    lastTrack: null,
+    userInfo: {
+      fullname: '', // TODO get this info,
+      followersCount: -1,
+      followingCount: -1,
+      followsYou: false,
+      avatar: null,
+      location: '' // TODO: get this information
+    }
   }
 }
 
-export default function (state: State = initialState, action: Action): State {
+// Just mutating the substate of the tracker
+function updateUserState (rootState: State, trackerState: TrackerState, action: Action) {
   switch (action.type) {
     case Constants.onFollowChecked:
       if (action.payload == null) {
-        return state
+        return rootState
```

Comment:
if this thing does 'nothing' it returns the root state, teh caller detects this and bails vs making duplicate copies. we should re-discuss immutable...